From df0218611b8f27ccaa924b8f88560c2e81966823 Mon Sep 17 00:00:00 2001
From: Daniel-Constantin Mierla <miconda@gmail.com>
Date: Thu, 2 Jan 2014 11:25:56 +0100
Subject: [PATCH] tmrec: updated the examples with tmrec duration

- hint to read rfc2445 for duration format

(cherry picked from commit 99d1170469c60206992c083f2ba0e9dc6b580d37)
---
 modules/tmrec/README              | 8 ++++----
 modules/tmrec/doc/tmrec_admin.xml | 7 ++++---
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/modules/tmrec/README b/modules/tmrec/README
index 6cffee0..84b7a56 100644
--- a/modules/tmrec/README
+++ b/modules/tmrec/README
@@ -138,7 +138,7 @@ modparam("tmrec", "separator", ";")
        subsequent intervals do not overlap. For non-recurring intervals,
        durations of any positive length are permitted. Zero-length
        duration means "forever". Negative-length durations are not
-       allowed.
+       allowed. See RFC 2445 for the format of duration.
      * frequency - can be one of the following values: "daily" - specify
        repeating periods based on an interval of a day or more; "weekly" -
        specify repeating periods based on an interval of a week or more;
@@ -222,9 +222,9 @@ modparam("tmrec", "separator", ";")
 
    Example 1.2. tmrec_match usage
 ...
-        if(tmrec_match("20120101T000000|24H|weekly|||SA,SU")
+        if(tmrec_match("20120101T000000|PT24H|weekly|||SA,SU")
         xdbg("it is weekend!\n");
-    if(tmrec_match("20120101T083000|10H|weekly|||MO,TU,WE,TH,FR")
+    if(tmrec_match("20120101T083000|PT10H|weekly|||MO,TU,WE,TH,FR")
         xdbg("it is with working hours\n");
 ...
 
@@ -298,7 +298,7 @@ if(is_leap_year("2010"))
 
    Example 1.4. time_period_match usage
 ...
-if(time_period_match("wd{1-5} hr{8-16}, wd{1-5} hr{17} min{0-29}"))
+if(time_period_match("wd{2-6} hr{8-16}, wd{1-5} hr{17} min{0-29}"))
         xdbg("Monday to Friday, 8:00 to 17:30\n");
 
 if(time_period_match("weekday { sat sun }, weekday {mo-fr} hr {17-8},wd{mo-wed}h
diff --git a/modules/tmrec/doc/tmrec_admin.xml b/modules/tmrec/doc/tmrec_admin.xml
index 782b1f1..7a429eb 100644
--- a/modules/tmrec/doc/tmrec_admin.xml
+++ b/modules/tmrec/doc/tmrec_admin.xml
@@ -127,7 +127,8 @@ modparam("tmrec", "separator", ";")
 			be small enough such that subsequent intervals do not overlap. 
 			For non-recurring intervals, durations of any positive length are 
 			permitted.  Zero-length duration means <quote>forever</quote>. 
-			Negative-length durations are not allowed.
+			Negative-length durations are not allowed. See RFC 2445 for
+			the format of duration.
 			</para>
 		</listitem>
 		<listitem>
@@ -274,9 +275,9 @@ modparam("tmrec", "separator", ";")
 		<title><function>tmrec_match</function> usage</title>
 		<programlisting format="linespecific">
 ...
-	if(tmrec_match("20120101T000000|24H|weekly|||SA,SU")
+	if(tmrec_match("20120101T000000|PT24H|weekly|||SA,SU")
         xdbg("it is weekend!\n");
-    if(tmrec_match("20120101T083000|10H|weekly|||MO,TU,WE,TH,FR")
+    if(tmrec_match("20120101T083000|PT10H|weekly|||MO,TU,WE,TH,FR")
         xdbg("it is with working hours\n");
 ...
 </programlisting>
-- 
1.8.5.2

